openid/src/main/java/fish/payara/security/openid/controller/ConfigurationController.java

```diff
@@ -365,6 +364,45 @@ private void validateConfiguration(OpenIdConfiguration configuration) {
         return errorMessages;
     }
 
+    /**
+     * Create Url query from pairs of parameters.
+     */
+    public String createUrlQuery(String[] parameters) {
+        StringBuilder extraParametersFromAnnotationBuf = new StringBuilder();
+        String extraParamDelim = "";
+        for (String extraParameter : parameters) {
+            String[] parts = extraParameter.split("=");
+            String key = parts[0];
+            String value = parts.length > 1 ? parts[1] : null;
+            extraParametersFromAnnotationBuf.append(extraParamDelim)
+                    .append(URLEncoder.encode(key, StandardCharsets.UTF_8))
+                    .append("=")
+                    .append(URLEncoder.encode(value, StandardCharsets.UTF_8));
+            extraParamDelim = "&";
+        }
+        return extraParametersFromAnnotationBuf.toString();
+    }
+
+    /**
+     * Parse Url query format to multimap.
+     */
+    private Map<String, List<String>> parseMultiMapFromUrlQuery(String query) {
+        Map<String, List<String>> multiMap = new LinkedHashMap<>();
+        String[] pairs = query.split("&");
+        for (String pair : pairs) {
+            String[] keyValue = pair.split("=");
+            String key = keyValue[0];
+            String value = keyValue.length > 1 ? URLDecoder.decode(keyValue[1], StandardCharsets.UTF_8) : null;
+            List<String> values = multiMap.get(key);
+            if (values == null) {
+                values = new ArrayList<>();
+                multiMap.put(key, values);
+            }
```

Comment:
Just nitpicking here: ` values = multiMap.computeIfAbsent(key, k -> new ArrayList<>())` -- I know it burned you before, but it is a useful method in single threaded code without null keys :)

And that reminds me... some test cases for invalid inputs like `" "` would be nice just to know what kind of exception they would throw.